### PR TITLE
GDPR use getConsentData CMP call to get consentString

### PR DESCRIPTION
**Pull request by jsnellbaker**
## Type of change
- [x] Bugfix

## Description of change
Previously the consentManagement module was using the CMP function `getVendorConsents` to obtain all aspects of the user's consent information.  It was pointed out in #2591 that this function is not the ideal one to use to obtain the user's consent information (specifically the encoded string).

This PR updates the logic of the module to use the `getConsentData` CMP function to grab the `consentString` and `gdprApplies` and then executes the `getVendorConsents` CMP function to grab the unparsed vendor consent information (to maintain backwards compatibility for that variable we offer to adapters).

### CODE REVIEW

File: modules/consentManagement.js [hunk 1]
```diff
@@ -42,27 +42,43 @@ const cmpCallMap = {
...
   let cmpCallbacks;
+  let cmpResponse = {};
+
+  // to collect the consent information from the user, we perform two calls to the CMP:
+  // first to collect the user's consent choices represented in an encoded string (via getConsentData)
+  // second to collect the user's full unparsed consent information (via getVendorConsents)
+  callIabCMP('getConsentData', function (consentResponse) {
+    cmpResponse.getConsentData = consentResponse;
+
+    callIabCMP('getVendorConsents', function (consentResponse) {
```
Comment (mkendall07) [inline modules/consentManagement.js]: why are you doing this serially and not in parallel ?

Comment (dbemiller) [reply modules/consentManagement.js]: I can chat about this with you if you're interested.

Comment (snapwich) [reply modules/consentManagement.js]: yeah I think parallel would be ideal, especially if these are making ajax requests.  If they're not, it probably doesn't matter.

Comment (mkendall07) [reply modules/consentManagement.js]: I don't think these are making ajax requests - so it probably doesn't matter so much.

**Comment by jsnellbaker** [comment]
I have updated the logic to have the CMP calls execute in a parallel instead of serial as well as resolve the current conflicts with master.  Please take another look and let me know of any additional feedback/suggestions.

Thanks.

**Comment by jsnellbaker** [comment]
Thanks for the approvals - merging in now.

**Pull request merged**